src/main/java/seedu/address/commons/util/StringUtil.java

```diff
@@ -120,7 +127,9 @@ public static String getFirstNWordsWithEllipsis(String string, int numOfWords) {
      */
     public static long getNumberOfWords(String string) {
         return Stream.ofNullable(string)
-                .flatMap(text -> Arrays.stream(text.split("\\s+")))
+                .map(text -> text.split("\\s+"))
+                .flatMap(Arrays::stream)
+                .filter(word -> word.length() > 0)
                 .count();
```

Comment:
Since we're trimming each word already maybe this simpler method can work
```java
Stream<String> words(String s) {
    if (s.trim().isEmpty()) return Stream.empty(); // or Stream.of()
    return Stream.of(s.split("\\s+"));
}

String firstNWordsWithEllipsis(String s, int n) {
    return words(s).limit(n).collect(Collectors.joining(" ")) + words(s).count() > n ? "..." : "";      
}
```